openfisca_core/projectors/helpers.py

```diff
@@ -10,25 +16,123 @@ def projectable(function):
     return function
 
 
-def get_projector_from_shortcut(population, shortcut, parent=None):
-    if population.entity.is_person:
-        if shortcut in population.simulation.populations:
-            entity_2 = population.simulation.populations[shortcut]
-            return projectors.EntityToPersonProjector(entity_2, parent)
-    else:
-        if shortcut == "first_person":
-            return projectors.FirstPersonToEntityProjector(population, parent)
-        role = next(
+def get_projector_from_shortcut(
+    population: Population | GroupPopulation,
+    shortcut: str,
+    parent: projectors.Projector | None = None,
+) -> projectors.Projector | None:
+    """???.
```

Comment:
It is used to project person or collective entity variables on collective entity and persons.
The first case is entity -> person, the other is person -> entity by choosing which role give ots individual variable to be projected onto the entity.

Exemples:
- for the first case: person.household("rent") assign a rent to every person
- the other case: family.first_parent("age")  (famille.demandeur for the french case or famille.conjoint).

Reply:
So depending on the type of population, and the roles and subroles in the case of a group population, it can be either a hard-coded value `first_person`, an entity `key`, or a unique role's key (a subrole).

Reply:
(Or a `role` with `max = 1`, I see now why my othjer PR is failing).